samza-core/src/main/java/org/apache/samza/system/inmemory/InMemorySystemProducer.java

```diff
@@ -95,6 +96,15 @@ public void send(String source, OutgoingMessageEnvelope envelope) {
     memoryManager.put(ssp, key, message);
   }
 
+  /**
+   * Populates the IME to the ssp configured, this gives user more control to set up Test environment
+   *
+   * @param envelope
+   */
+  public void send(IncomingMessageEnvelope envelope) {
```

Comment:
1. More docs on the parameter or remove it.
2. Can we clarify that this is not intentional for production use?
3. `Populates the IME to the ssp configured` seems a little confusing. I'd explicitly call out that the method takes payloads wrapped in Samza internal representation and clients need to adhere to the semantics IME. e.g. populating the offset field and order in which put is invoked so that IMEs has monotonically increasing offsets.

Reply:
Added more docs, documented not to be used for production use cases, added additional checks to fail if offset policy does not adhere.

Reply:
Thanks! Can you also add it is not thread safe?